R/eml.R

```diff
@@ -416,12 +416,19 @@ eml_party <- function(type="associatedParty",
 
   # Role
   if (!is.null(role)) {
-    if (type != "associatedParty") {
+    # Only allow roles to be set if type is associatedParty or personnel
+    if (type != "associatedParty" && type != "personnel") {
       stop(call. = FALSE,
-           paste0("Setting a role is only valid on an associatedParty, not a ", type, "."))
+           paste0("Setting a role is only valid on an associatedParty or personnel, not a ", type, "."))
+    }
+    
+    # If type is personnel, role needs to be ListOfrole, otherwise just role
+    if (type == "personnel") {
+      roles <- lapply(role, function(x) { r <- new("role"); r@.Data <- x; r } )
+      party@role <- new("ListOfrole", roles)
+    } else {
+      party@role <- new("role", .Data = role)
```

Comment:
More and more, I'm trying to use coercions instead of constructing objects using `new`, like `as(role, "role")`